### DISCUSSION TITLE
Running reverse-proxy Nginx server and Wiki (docker) on the same machine

### DISCUSSION THREAD

**Discussion by MotikaCraft-TheJoshua** [requarks/wiki#3845] [Help / Questions]
Hello.
I followed this installation process: https://docs.requarks.io/install/ubuntu
Now, i have installed the wiki, everything works (perfectly).
The only thing is that I cannot run my website which is uses the same port - NGINX uses it
This is what i get when i do `docker start wiki`:
```
root@vps:~# docker start wiki
Error response from daemon: driver failed programming external connectivity on endpoint wiki (xxxxx):
Error starting userland proxy: listen tcp4 0.0.0.0:443: bind: address already in use
```
I'd like to run primary website as "exampledomain.com" and wiki site as "wiki.exampledomain.com".
I hope someone can help me solve this, even though I have already installed wiki server, so yeah, I don't know if I should change docker's port and how to change it because installation is done and how should I change port where the process of installation done it, don't know where certainly to find everything (I hope this last sentence isn't that confusing)

Thank you in advance.

**Comment by MotikaCraft-TheJoshua** [comment]
![image](https://user-images.githubusercontent.com/56804505/114936885-54554480-9e3d-11eb-89a8-65c9908a8719.png)

I see this but then I don't understand why it says this
![image](https://user-images.githubusercontent.com/56804505/114936992-7e0e6b80-9e3d-11eb-87f8-925558ebafff.png)

**Comment by Phil-hostwiki** [comment]
I think you should create Nginx reverse proxy block for both Wiki.js and your primary site.

**Reply by MotikaCraft-TheJoshua** [reply]
Can you tell me what would nginx block for wiki.js look alike? I already have nginx block for my primary site (example.com).

**Reply by MotikaCraft-TheJoshua** [reply]
and what should i do about that i already done that docker/ubuntu installation guide - I installed wiki.js
should i just ignore that and start all over - configure wikijs to work with nginx?

**Reply by Phil-hostwiki** [reply]
The Nginx block should look something like this assuming Wiki.js is running on port 3000.
Then use certbot to generate SSL. 

```
server {
  listen 80;
  listen [::]:80;

  server_name wiki.example.com;

  location / {
      proxy_pass http://localhost:3000/;
      proxy_http_version 1.1;
      client_max_body_size 256M;
      proxy_set_header Connection "upgrade";
      proxy_set_header Host $http_host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
  }
}
```

**Reply by MotikaCraft-TheJoshua** [reply]
> The Nginx block should look something like this assuming Wiki.js is running on port 3000.
> Then use certbot to generate SSL.
> 
> ```
> server {
>   listen 80;
>   listen [::]:80;
> 
>   server_name wiki.example.com;
> 
>   location / {
>       proxy_pass http://localhost:3000/;
>       proxy_http_version 1.1;
>       client_max_body_size 256M;
>       proxy_set_header Connection "upgrade";
>       proxy_set_header Host $http_host;
>       proxy_set_header X-Real-IP $remote_addr;
>       proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
>       proxy_set_header X-Forwarded-Proto $scheme;
>   }
> }
> ```

what about 443?

**Reply by Phil-hostwiki** [reply]
If you use certbot, it'll handle that.

**Reply by MotikaCraft-TheJoshua** [reply]
Alright, but, i need to download wikijs once again and start it with nodejs?

**Reply by Phil-hostwiki** [reply]
You should re-install it without the https option as Nginx and Certbot will handle that

https://docs.requarks.io/install/ubuntu

**Reply by MotikaCraft-TheJoshua** [reply]
> You should re-install it without the https option as Nginx and Certbot will handle that
> 
> https://docs.requarks.io/install/ubuntu

alright I will do it.
Any idea how to properly uninstall it and install it again? as that would
be a re-install.

**Reply by Phil-hostwiki** [reply]
```
docker stop wiki
docker rm wiki

docker create --name=wiki -e DB_TYPE=postgres -e DB_HOST=db -e DB_PORT=5432 -e DB_PASS_FILE=/etc/wiki/.db-secret -v /etc/wiki/.db-secret:/etc/wiki/.db-secret:ro -e DB_USER=wiki -e DB_NAME=wiki -e UPGRADE_COMPANION=1 --restart=unless-stopped -h wiki --network=wikinet -p 80:3000 -p 443:3443 requarks/wiki:2

```

**Reply by MotikaCraft-TheJoshua** [reply]
> ```
> docker stop wiki
> docker rm wiki
> 
> docker create --name=wiki -e DB_TYPE=postgres -e DB_HOST=db -e DB_PORT=5432 -e DB_PASS_FILE=/etc/wiki/.db-secret -v /etc/wiki/.db-secret:/etc/wiki/.db-secret:ro -e DB_USER=wiki -e DB_NAME=wiki -e UPGRADE_COMPANION=1 --restart=unless-stopped -h wiki --network=wikinet -p 80:3000 -p 443:3443 requarks/wiki:2
> ```

So that will basically reinstall it? Will also remove the installed https option?

**Reply by MotikaCraft-TheJoshua** [reply]
Also, after that, I do `docker start wiki`?

**Reply by Phil-hostwiki** [reply]
Yes, start the container.

**Reply by MotikaCraft-TheJoshua** [reply]
> Yes, start the container.

When i do `docker start wiki`, I get
```
root@xxxxx:~# docker start wiki
Error response from daemon: driver failed programming external connectivity on endpoint wiki (xxxxx): Error starting userland proxy: listen tcp4 0.0.0.0:443: bind: address already in use
Error: failed to start containers: wiki
```

**Reply by MotikaCraft-TheJoshua** [reply]
The same as main image about this post
![image](https://user-images.githubusercontent.com/56804505/115026321-a68c7900-9ec2-11eb-9bf9-3a2548d2b3a6.png)

**Comment by Phil-hostwiki** [comment]
I'm not sure if you have fixed the issue but this will solve it.

When creating the container, replace `-p 80:3000 -p 443:3443` with `-p 3000:3000`

Wiki.js will run on port 3000. 
Then use the Nginx reverse proxy on https://github.com/Requarks/wiki/discussions/3845#discussioncomment-617949
